File: CS/DxSample/Gallery/CustomGallery.cs
using DevExpress.XtraBars.Ribbon.Gallery;
using DevExpress.XtraBars.Ribbon.ViewInfo;

namespace DxSample.Gallery {
    public class CustomGallery :GalleryControlGallery {
        public CustomGallery() : base() { }
        public CustomGallery(CustomGalleryControl galleryControl) : base(galleryControl) { }

        protected override BaseGalleryViewInfo CreateViewInfo() {
            return new CustomGalleryViewInfo(this);
        }
    }
}

File: VB/DxSample/Gallery/CustomGallery.vb
Imports DevExpress.XtraBars.Ribbon.Gallery
Imports DevExpress.XtraBars.Ribbon.ViewInfo

Namespace DxSample.Gallery

    Public Class CustomGallery
        Inherits GalleryControlGallery

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal galleryControl As CustomGalleryControl)
            MyBase.New(galleryControl)
        End Sub

        Protected Overrides Function CreateViewInfo() As BaseGalleryViewInfo
            Return New CustomGalleryViewInfo(Me)
        End Function
    End Class
End Namespace
